eng/tools/generator/cmd/v2/release/releaseCmd.go

```diff
@@ -99,49 +98,20 @@ type commandContext struct {
 }
 
 func (c *commandContext) execute(sdkRepoParam, specRepoParam string) error {
-	var err error
-	var sdkRepo repo.SDKRepository
-	// create sdk and spec git repo ref
-	if commitIDRegex.Match([]byte(sdkRepoParam)) {
-		sdkRepo, err = repo.CloneSDKRepository(c.flags.SDKRepo, sdkRepoParam)
-		if err != nil {
-			return fmt.Errorf("failed to get sdk repo: %+v", err)
-		}
-	} else {
-		path, err := filepath.Abs(sdkRepoParam)
-		if err != nil {
-			return fmt.Errorf("failed to get the directory of azure-sdk-for-go: %v", err)
-		}
-
-		sdkRepo, err = repo.OpenSDKRepository(path)
-		if err != nil {
-			return fmt.Errorf("failed to get sdk repo: %+v", err)
-		}
+	sdkRepo, err := common.GetSDKRepo(sdkRepoParam, c.flags.SDKRepo)
+	if err != nil {
+		return err
 	}
 
-	specCommitHash := ""
-	if commitIDRegex.Match([]byte(specRepoParam)) {
-		specCommitHash = specRepoParam
-	} else {
-		path, err := filepath.Abs(specRepoParam)
-		if err != nil {
-			return fmt.Errorf("failed to get the directory of azure-rest-api-specs: %v", err)
-		}
-		specRepo, err := repo.OpenSpecRepository(path)
-		if err != nil {
-			return fmt.Errorf("failed to get spec repo: %+v", err)
-		}
-		specHeader, err := specRepo.Head()
-		if err != nil {
-			return fmt.Errorf("failed to get HEAD ref of azure-rest-api-specs: %+v", err)
-		}
-		specCommitHash = specHeader.Hash().String()
+	specCommitHash, err := common.GetSpecCommit(specRepoParam)
+	if err != nil {
+		return err
 	}
 
 	log.Printf("Release generation for rp: %s, namespace: %s", c.rpName, c.namespaceName)
 	generateCtx := common.GenerateContext{
-		SDKPath:        sdkRepo.Root(),
-		SDKRepo:        &sdkRepo,
+		SDKPath:        (*sdkRepo).Root(),
```

Comment:
Why we have to dereference this variable? this looks weird

Reply:
Changed.